pallets/wnodl/src/lib.rs

```diff
@@ -0,0 +1,197 @@
+/// Learn more about FRAME and the core library of Substrate FRAME pallets:
+/// <https://docs.substrate.io/v3/runtime/frame>
+pub use pallet::*;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+
+#[frame_support::pallet]
+pub mod pallet {
+    pub use ethereum_types::Address as EthAddress;
+    use frame_support::{
+        pallet_prelude::*,
+        traits::{Contains, Currency, LockableCurrency},
+    };
+    use frame_system::pallet_prelude::*;
+    pub use sp_core::H256 as EthTxHash;
+    use sp_runtime::traits::{CheckedAdd, Saturating, Zero};
+
+    pub type CurrencyOf<T> = <T as Config>::Currency;
+    pub type BalanceOf<T> =
+        <<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config {
+        type Currency: LockableCurrency<Self::AccountId, Moment = Self::BlockNumber>;
+
+        type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+        /// Trusted bots/oracles which can settle funds after wrapping is initiated
+        type Oracles: Contains<Self::AccountId>;
+
+        /// The customers who've gone under the KYC process and are eligible to wrap their Nodls.  
+        type KnownCustomers: Contains<Self::AccountId>;
+    }
+
+    #[pallet::pallet]
+    #[pallet::generate_store(pub(super) trait Store)]
+    pub struct Pallet<T>(_);
+
+    #[pallet::storage]
+    #[pallet::getter(fn total_initiated)]
+    /// The sum of wNodl funds that is initiated by this pallet so far.
+    pub type TotalInitiated<T: Config> = StorageValue<_, BalanceOf<T>>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn total_settled)]
+    /// The sum of wNodl funds that is settled by this pallet so far.
+    pub type TotalSettled<T: Config> = StorageValue<_, BalanceOf<T>>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn balances)]
+    /// The amount of initiated and settled wnodl for an account id.
+    /// NOTE: keeping the trace of the jobs can be done fullly offchain through our oracle wNodl-bot
+    /// and by monitoring the events. We will however keep them here for our customers convenienve
+    /// This would make sense to create a custom rpc and an off-chain storage for this purpose later.
+    pub type Balances<T: Config> =
+        StorageMap<_, Twox64Concat, T::AccountId, (BalanceOf<T>, BalanceOf<T>)>;
+
+    #[cfg(feature = "runtime-benchmarks")]
+    #[pallet::storage]
+    #[pallet::getter(fn benchmark_known_customers)]
+    /// An internally kept list for the benchmark tests.
+    pub type WhitelistedCallers<T: Config> = StorageValue<_, Vec<T::AccountId>>;
+
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    pub enum Event<T: Config> {
+        /// Wrapping Nodl is initiated
+        /// parameters. [account's address on Nodle chain, amount of Nodl fund, destination address on Ethereum main-net]
```

Comment:
I think we need a `\` before the paranthesis

Reply:
Do you mean like this:
`/// Parameters: \[account's address on Nodle chain, amount of Nodl fund, destination address on Ethereum main-net\]`
?

Reply:
Done